### PR TITLE
Move random index generator from p2p to crypto

### PR DESCRIPTION
**Pull request by IPGlider**

### CODE REVIEW

File: src/crypto/crypto.h [hunk 1]
```diff
@@ -152,6 +152,20 @@ namespace crypto {
...
+  /* Generate an index between 0 and a given value with fixed probability
+   */
+  inline size_t random_index_with_fixed_probability(size_t max_index)
+  {
+    //divide by zero workaround
+    if(!max_index)
+      return 0;
+
+    size_t x = crypto::rand<size_t>()%(max_index+1);
+    size_t res = (x*x*x)/(max_index*max_index); //parabola \/
```
Comment (moneromooo-monero) [inline src/crypto/crypto.h]: Integer overflow on most of the range, as a side note. But since it's just moving around, ok.

**Comment by iamsmooth** [comment]
Do no think this is a good idea to move without more clearly documenting what this does and why it is implemented in the manner it is. The code looks fishy as @moneromooo-monero mentioned and it would be a bad idea to encourage it to be used elsewhere without better packaging (and testing)

**Comment by IPGlider** [comment]
@moneromooo-monero @iamsmooth 

I agree with both of you.

I have been doing some calculations, and I think that using this method increases the chances of the result being ```0```, and makes every value of ```x``` smaller.

For example with a ```max_input = 3500``` every ``` x < 50 2^(1/3) 7^(2/3)``` then ```res = 0```.

Aprox ```x = 230.52``` hence any ```x <= 230.52``` will output a ```res = 0```.

https://www.wolframalpha.com/input/?i=xˆ3%2F3500ˆ2+%3D+1

A plot with ```max_inputs = 3500```: https://www.wolframalpha.com/input/?i=plot+xˆ3%2F3500ˆ2+x%3D0+to+3500

Thus it is my understanding that it would be better to just use ```crypto::rand<size_t>()%(max_index+1)``` and remove the function completely.

If you need any further justification to not use this function let me know and I will try to provide more evidence.

Feel free to check my math and let me know if anything doesn't seem right.

**Comment by iamsmooth** [comment]
Presumably there might be some reason to prefer lower numbers here? The index is into some list; is the list sorted?

**Comment by IPGlider** [comment]
Will look deeper into into it over the week.

**Comment by NanoAkron** [comment]
Should this one be closed in favour of merging #1687 first, then reconsidering?

**Comment by IPGlider** [comment]
Yes @NanoAkron this should **not** be merged. I don't know if this should be closed and open an issue with the investigation about how the method works.

**Comment by moneromooo-monero** [comment]
I wouldn't change to a uniform distribution without understanding why cubic was used in the first place.

**Pull request closed without merging**